### PR TITLE
Add Durations

### PR DESCRIPTION
**Pull request by exallium**
Durations is a small library to make durations stricter.  Please see README for details.

### CODE REVIEW

**Comment by exallium** [comment]
Added CompareTo Functionality, I'll write a test for this tomorrow.

**Comment by exallium** [comment]
@markmckenna This is ready to be looked at.

File: durations/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+Durations splits different time divisions into explicit subtypes of a sealed class `Duration`
+
+E.G.
+
+```
+
+interface MyApi {
+
+    // Function's signature explicitly declares that it will utilize milliseconds
+    fun action(milliseconds: Duration.Milliseconds)
```
Comment (markmckenna) [inline durations/README.md]: Prop name -> `duration`?

File: durations/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Durations -- Typesafe Time Units
+
+Author: Alex Hart (exallium)
```
Comment (markmckenna) [inline durations/README.md]: Link to your Github/etc. account?

File: durations/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+fun run(myApi: MyApi) {
+
+  val ms = milliseconds(1000) // Returns a Duration.Milliseconds
+  val s = seconds(10) // Returns a Duration.Seconds
+
+  myApi.action(ms) // Works great!
+
+  // myApi.action(s) // Error!
+
+  myApi.action2(s + ms) // Works! + and - will return a Duration
```
Comment (markmckenna) [inline durations/README.md]: But `myApi.action(s + ms)` won't work, correct?  Is there a way for it to work?  Is there a practical reason why someone would specify `Duration.Milliseconds` in their accepting type signature?

**Review by markmckenna — APPROVED** [review]

File: durations/README.md (hunk 3 above)
Comment (exallium) [reply durations/README.md]: I was thinking about this.  I don't think there's ever a reason to require a specific unit.  I'll update the test documentation to remove `action1`

File: durations/README.md (hunk 1 above)
Comment (exallium) [reply durations/README.md]: method removed

File: durations/README.md (hunk 2 above)
Comment (exallium) [reply durations/README.md]: linked

**Pull request merged**